### PR TITLE
Better entities parsing

### PR DESCRIPTION
**Pull request by nevrome**
I tried to improve the entities list parsing, not least in reaction to #139. I hope it works and fails (!) now as expected.

### CODE REVIEW

File: test/Poseidon/EntitiesListSpec.hs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    let g1 = "test/testDat/testEntityFiles/goodEntities1.txt"
+        g2 = "test/testDat/testEntityFiles/goodEntities2.txt"
+        b1 = "test/testDat/testEntityFiles/badEntities1.txt"
+    it "should parse good, single-value-per-line files correctly" $ do
+        g1res <- readEntitiesFromFile g1
+        g1res `shouldBe` [Ind "a", Group "b", Pac "c"]
+    it "should parse good, multi-value-per-line files correctly" $ do
+        g2res <- readEntitiesFromFile g2
+        g2res `shouldBe` [Ind "a1", Ind "a2", Group "b1", Pac "c1", Pac "c2", Group "b2", Group "b3"]
+    it "should fail to parse bad files and throw an exception" $ do
```
Comment (stschiff) [inline test/Poseidon/EntitiesListSpec.hs:48]: Do you also have a test for the issue raised in #139, so an empty line in the forge file? I couldn't see it in the new files added with this PR.

**Review by stschiff — COMMENTED** [review]
Very good to have all these tests. I made one comment...

File: test/Poseidon/EntitiesListSpec.hs (hunk 1 above)
Comment (nevrome) [reply test/Poseidon/EntitiesListSpec.hs:49]: The issue in #139 are additional spaces, not newlines. Empty lines are not allowed right now. I added a test case to reflect that. We could be more flexible here, of course :man_shrugging:. But I guess then we could also consider to add proper comments :thinking:

**Comment by nevrome** [comment]
Alright - I added support for empty lines and comments. Could you please take another quick look, @stschiff? It was a great parsec learning opportunity (!!), but maybe my solution is not to your standards :sweat_smile:
